checkers: flagName panics on `cmd/compile/internal/gc`

```
gocritic check -enable=flagName cmd/compile/internal/gc
flagName: error: runtime error: index out of range
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 1050 [running]:
github.com/go-lintpack/lintpack/linter/lintmain/internal/check.(*linter).checkFile.func1.1(0xc006345c60, 0xc0041ec480)
	$GOPATHsrc/github.com/go-lintpack/lintpack/linter/lintmain/internal/check/check.go:140 +0x15c
panic(0x825560, 0xced9b0)
	$GOROOT/src/runtime/panic.go:522 +0x1b5
github.com/go-critic/go-critic/checkers.(*flagNameChecker).VisitExpr(0xc000fdd1b0, 0x8ede40, 0xc003cf37c0)
	$GOPATHsrc/github.com/go-critic/go-critic/checkers/flagName_checker.go:46 +0x722
github.com/go-lintpack/lintpack/astwalk.(*exprWalker).WalkFile.func1(0x8e8b80, 0xc003cf37c0, 0x1)
	$GOPATHsrc/github.com/go-lintpack/lintpack/astwalk/expr_walker.go:23 +0x7f
go/ast.inspector.Visit(0xc001051e10, 0x8e8b80, 0xc003cf37c0, 0x8e7d40, 0x52ea0c)
	$GOROOT/src/go/ast/walk.go:373 +0x3a
go/ast.Walk(0x8e7d40, 0xc001051e10, 0x8e8b80, 0xc003cf37c0)
	$GOROOT/src/go/ast/walk.go:52 +0x66
go/ast.Walk(0x8e7d40, 0xc001051e10, 0x8e8ac0, 0xc003d0d560)
	$GOROOT/src/go/ast/walk.go:146 +0x183b
go/ast.Walk(0x8e7d40, 0xc001051e10, 0x8e9100, 0xc003cf3800)
	$GOROOT/src/go/ast/walk.go:230 +0x1c3b
go/ast.walkStmtList(0x8e7d40, 0xc001051e10, 0xc003989200, 0x7, 0x8)
	$GOROOT/src/go/ast/walk.go:32 +0x9e
go/ast.Walk(0x8e7d40, 0xc001051e10, 0x8e8b00, 0xc003d0d9e0)
	$GOROOT/src/go/ast/walk.go:224 +0x1b67
go/ast.Walk(0x8e7d40, 0xc001051e10, 0x8e9380, 0xc003c5fc20)
	$GOROOT/src/go/ast/walk.go:285 +0x2961
go/ast.walkStmtList(0x8e7d40, 0xc001051e10, 0xc0039a6800, 0xb, 0x10)
	$GOROOT/src/go/ast/walk.go:32 +0x9e
go/ast.Walk(0x8e7d40, 0xc001051e10, 0x8e8b00, 0xc003d0db30)
	$GOROOT/src/go/ast/walk.go:224 +0x1b67
go/ast.Walk(0x8e7d40, 0xc001051e10, 0x8e8f80, 0xc003d0db60)
	$GOROOT/src/go/ast/walk.go:344 +0xd73
go/ast.Inspect(...)
	$GOROOT/src/go/ast/walk.go:385
github.com/go-lintpack/lintpack/astwalk.(*exprWalker).WalkFile(0xc000fdd220, 0xc003989400)
	$GOPATHsrc/github.com/go-lintpack/lintpack/astwalk/expr_walker.go:21 +0x12d
github.com/go-lintpack/lintpack.(*Checker).Check(...)
	$GOPATHsrc/github.com/go-lintpack/lintpack/lintpack.go:126
github.com/go-lintpack/lintpack/linter/lintmain/internal/check.(*linter).checkFile.func1(0xc006345c60, 0xc003989400, 0xc0062af480, 0x1, 0x1, 0x0, 0xc0041ec480)
	$GOPATHsrc/github.com/go-lintpack/lintpack/linter/lintmain/internal/check/check.go:148 +0x99
created by github.com/go-lintpack/lintpack/linter/lintmain/internal/check.(*linter).checkFile
	$GOPATHsrc/github.com/go-lintpack/lintpack/linter/lintmain/internal/check/check.go:130 +0x120
```

--- a/checkers/flagName_checker.go
+++ b/checkers/flagName_checker.go
@@ -3,6 +3,7 @@ package checkers
 import (
 	"go/ast"
 	"go/constant"
+	"go/types"
 	"strings"
 
 	"github.com/go-lintpack/lintpack"
@@ -30,13 +31,14 @@ type flagNameChecker struct {
 
 func (c *flagNameChecker) VisitExpr(expr ast.Expr) {
 	call := astcast.ToCallExpr(expr)
-	sym := astcast.ToIdent(astcast.ToSelectorExpr(call.Fun).Sel)
-	obj := c.ctx.TypesInfo.ObjectOf(sym)
-	if obj == nil {
+	calledExpr := astcast.ToSelectorExpr(call.Fun)
+	obj, ok := c.ctx.TypesInfo.ObjectOf(astcast.ToIdent(calledExpr.X)).(*types.PkgName)
+	if !ok {
 		return
 	}
-	pkg := obj.Pkg()
-	if !isStdlibPkg(pkg) || pkg.Name() != "flag" {
+	sym := calledExpr.Sel
+	pkg := obj.Imported()
+	if !isStdlibPkg(pkg) || obj.Name() != "flag" {
 		return
 	}
 

--- a/checkers/testdata/flagName/negative_tests.go
+++ b/checkers/testdata/flagName/negative_tests.go
@@ -4,6 +4,12 @@ import (
 	"flag"
 )
 
+func methodCall() {
+	// See #784.
+	var getter flag.Getter
+	_ = getter.String()
+}
+
 func noWhitespace() {
 	_ = flag.Bool("name", false, "")
 	_ = flag.Duration("name", 0, "")

